### PR TITLE
Feature/2467 Vectorize Discrete RNGs and a few others

### PR DESCRIPTION
**Pull request by roualdes**
#### Submission Checklist

- [ ] Run unit tests: `./runTests.py src/test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

Tests passed: `make clean && ./runTests.py -j3 src/test/unit/lang`

#### Summary

Expose (most) discrete and a few straggling continuous RNGs following PR https://github.com/stan-dev/math/pull/833 in a continued effort to fulfill https://github.com/stan-dev/stan/issues/2467.

#### Intended Effect

Specifically, expose the following RNGs

```
multi_normal_rng
multi_normal_cholesky_rng
multi_student_t_rng
bernoulli_rng
beta_binomial_rng
binomial_rng
neg_binomial_2_rng
neg_binomial_2_log_rng
neg_binomial_rng
poisson_rng
poisson_log_rng
uniform_rng
von_mises_rng
```

#### How to Verify

Tests in `src/test/test-models/good/function-signatures/distributions/rngs.stan`

#### Side Effects

None intended.

#### Documentation

In `src/docs/stan-reference/distributions.tex`

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

California State University, Chico

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by bob-carpenter** [comment]
@roualdes  There's a conflict here that needs to get resolved.  I updated all the doc from under your issue for the 2.18 release.  The merge conflict should be minor as not that much changed in the functions doc.

File: src/stan/lang/function_signatures.h [hunk 1]
```diff
@@ -110,7 +112,13 @@ for (size_t i = 0; i < int_vector_types.size(); i++)
             int_vector_types[i], int_vector_types[j],
 	    vector_types[k], vector_types[l]);
       }
-add("beta_binomial_rng", expr_type(int_type()), expr_type(int_type()), expr_type(double_type()), expr_type(double_type()));
+for (const auto& t : int_vector_types) {
```
Comment (bob-carpenter) [inline src/stan/lang/function_signatures.h]: Nice!  We really need to bring the rest of our code up to C++11 standards.

**Comment by roualdes** [comment]
Just attempted some git fu.  Hoping for the best.  Thanks for the quick review.

**Comment by roualdes** [comment]
Looks like one of the jobs timed out.

**Comment by bbbales2** [comment]
@roualdes I just restarted it but that's probably not gonna help anything. @seantalts ideas?

**Comment by seantalts** [comment]
Travis timeouts usually are fixed by just restarting that particular job. Fingers crossed!

**Pull request merged**